controllers/storagecluster/kms_resources.go

```diff
@@ -35,10 +37,51 @@ var (
 type kmsConfigMapValidateFunc func(*corev1.ConfigMap) error
 
 var (
-	// just to make sure 'isKMSProviderReachable' function follows the validate function type
+	// just to make sure 'reachKMSProvider' function follows the validate function type
 	_ kmsConfigMapValidateFunc = reachKMSProvider
 )
 
+type kmsResourceManager struct{}
+
+// ensureCreated a dummy function to make 'kmsResourceManager' compliant to 'resourceManager' interface
+func (kmsRMgr *kmsResourceManager) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	return nil
+}
+
+func (kmsRMgr *kmsResourceManager) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	// if 'KMS' is not enabled, nothing to delete
+	if !sc.Spec.Encryption.KeyManagementService.Enable {
+		return nil
+	}
+	var ensureDeleteErrors []error
+	var allErrors []error = make([]error, 2)
+	var allRuntimeObjs []runtime.Object = make([]runtime.Object, 2)
+	allRuntimeObjs[0], allErrors[0] = getKMSConfigMap(sc, r.Client, nil)
+	allRuntimeObjs[1], allErrors[1] = getKMSSecretToken(sc, r.Client)
+	allKMSResourceNames := []string{KMSConfigMapName, KMSTokenSecretName}
+
+	for idx, err := range allErrors {
+		errLog := "while retrieving"
+		if err == nil {
+			errLog = "while deleting"
+			err = r.Client.Delete(context.TODO(), allRuntimeObjs[idx])
+		}
+		if err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, fmt.Sprintf("Error occurred %v the kms resource: %v", errLog, allKMSResourceNames[idx]))
+			ensureDeleteErrors = append(ensureDeleteErrors, err)
+		}
+	}
+	// throw the first error happened
+	// TODO: wrap all the errors and return a single error
+	for _, err := range ensureDeleteErrors {
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Is there any (simple) way I can consolidate all the errors and send out a single error? I thought of many complicated approaches... (=
@iamniting , @raghavendra-talur , @jarrpa ^^^

Reply:
As of today, we are not going ahead in any of `ensureDeleted` func's if there is any err. We are returning back from the first error itself. @raghavendra-talur What do you think about the above approach? Where we are deleting everything and consolidate errors if any.

Reply:
If `A` depends on `B`, yes, we cannot delete `A` until we delete `B`. But in this case, `A` and `B` are two independent items/resources, that's why I was trying to delete all of them, no matter whether the other one fails (or not).

PS: this was one of the old times irritating case of `Ctrl+A` delete of large sets of files in `some OS-es`, where the `Delete` fails at the first occurrence of any corrupted file and we have to again unselect the corrupted file (one-by-one) and do the process all over again (and again). (=